File: Python/ProjectEuler16.py
import math


first = 2
last = math.pow(first,1000)

sumArythmeticSeq = lambda a1,an,n: ((a1 + an) / 2) * n
#data = list(int(sumArythmeticSeq(first,last,1000)))

data = [int(x) for x in list(str(int(sumArythmeticSeq(first,last,1000))))]

print(sum(data,0)) 

File: TypeScript/ProjectEuler16.ts
interface ISumOfArythmeticSeqFunc<T> {
    (data: T): number
}
class Euler16Data {
    get first(){
        return this._first;
    }

    get last() {
        return this._last;
    }

    get quantity() {
        return this._count;
    }

    constructor(private _first, private _last, private _count) {
    }
}

class Euler16 {
    
    public sumArythmeticSeq = (func: ISumOfArythmeticSeqFunc<Euler16Data>) => {
        return (data: Euler16Data) => {
            return func(data);
        };
    }


}

var last = Math.pow(2, 1000);
var func: ISumOfArythmeticSeqFunc<Euler16Data> = (data) => ((data.first + data.last) / 2) * data.quantity;
var euler16Data = new Euler16Data(2, last, 1000);
var result = new Euler16().sumArythmeticSeq(func);
console.log(result(euler16Data).toString().split("").reduce((x,y) => x + y));
while (true);
